lib/srv/heartbeatv2.go

```diff
@@ -73,20 +74,32 @@ func NewSSHServerHeartbeat(cfg SSHServerHeartbeatConfig) (*HeartbeatV2, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	var metadataPtr atomic.Pointer[metadata.Metadata]
 	inner := &sshServerHeartbeatV2{
-		getServer: func(ctx context.Context) *types.ServerV2 {
-			server := cfg.GetServer()
-			metadata, err := metadata.Get(ctx)
-			if err == nil {
-				if metadata.CloudMetadata != nil {
-					server.SetCloudMetadata(metadata.CloudMetadata)
+		getMetadata: metadata.Get,
+		announcer:   cfg.Announcer,
+	}
+	inner.getServer = func(ctx context.Context) *types.ServerV2 {
+		server := cfg.GetServer()
+
+		meta := metadataPtr.Load()
+		if meta == nil {
```

Comment:
nit: Use the smaller scope:

```suggestion
		if meta := metadataPtr.Load(); meta == nil {
```